src/components/App.tsx

```diff
@@ -107,9 +107,13 @@ import { actionFinalize } from "../actions";
 
 function withBatchedUpdates<
   TFunction extends ((event: any) => void) | (() => void)
->(func: TFunction) {
+>(
+  func: Parameters<TFunction>["length"] extends 0 | 1
```

Comment:
mindblown.gif

Reply:
Me too, courtesy of folks on TS gitter. TS is insane :).